dotnet/src/Experimental/Agents/AgentMessage.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Threading;
+using Microsoft.SemanticKernel.ChatCompletion;
+
+namespace Microsoft.SemanticKernel.Experimental.Agents;
+
+/// <summary>
+/// Represents agent message used by agents inherited from <see cref="KernelAgent" /> class.
+/// </summary>
+public class AgentMessage
```

Comment:
Any thought on the "cititation" requirement we discussed?

Reply:
I believe it can be easily modeled by another subclass of KernelType; however, we need to confirm that this is the case.